### ISSUE TITLE
Deleted collectives should not prevent slugs from being taken again

### ISSUE THREAD

**Issue by Betree** [opencollective/opencollective#2410]
**Current behavior**
When you delete a collective, we set `deletedAt` but don't change the slug. As there's a constraint on the slug, it stays locked forever.

This is a problem because some collectives are deleting an organization created by mistake to re-create a collective instead and are blocked by the slug validation. (ex: https://opencollective.freshdesk.com/a/tickets/1521). As deleted collectives don't appear on Forest, support can't help such users.

**Proposed behavior**

Add `-deleted` to the slug when deleting collectives.

**Comment by piamancini** [comment]
this should also be the behaviour for the email address of profiles that can't be deleted because they have transactions and therefore are anonymized. 
Separate issue for this in the future ...

**Comment by maddhruv** [comment]
At Line: `791` here - `server/graphql/v1/mutations/collectives.js`
it looks like the new collective slug after deletion is already something different than the one created with - 
```js
// deleteCollective
const newSlug = `${collective.slug}-${Date.now()}`;
return collective.update({ slug: newSlug });
```

But seems to be not doing this at `deleteEventCollective`
At Line: `486` here - `server/graphql/v1/mutations/collectives.js`

**Comment by Betree** [comment]
In this case it wasn't an event, it was an organization. But you're right that we already have something for it, I'll monitor similar cases in the future.
